File: app/src/main/java/com/app/ralaunch/utils/LocaleManager.java
package com.app.ralaunch.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Build;
import android.os.LocaleList;

import java.util.Locale;

/**
 * 多语言管理器
 *
 * 负责管理应用的语言设置和切换:
 * - 支持中文、英文等多种语言
 * - 持久化语言选择
 * - 动态切换语言无需重启应用
 * - 跟随系统语言选项
 */
public class LocaleManager {
    private static final String TAG = "LocaleManager";
    private static final String PREFS_NAME = "app_prefs";
    private static final String KEY_LANGUAGE = "app_language";

    // 支持的语言
    public static final String LANGUAGE_AUTO = "auto";      // 跟随系统
    public static final String LANGUAGE_ZH = "zh";          // 简体中文
    public static final String LANGUAGE_EN = "en";          // English
    public static final String LANGUAGE_RU = "ru";          // Русский
    
    // Locale 常量
    private static final Locale LOCALE_RUSSIAN = new Locale("ru");

    /**
     * 获取当前设置的语言
     * @param context Context
     * @return 语言代码 (auto/zh/en)
     */
    public static String getLanguage(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        return prefs.getString(KEY_LANGUAGE, LANGUAGE_AUTO);
    }

    /**
     * 设置语言
     * @param context Context
     * @param language 语言代码 (auto/zh/en)
     */
    public static void setLanguage(Context context, String language) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_LANGUAGE, language).apply();
    }

    /**
     * 应用语言设置到Context
     * @param context 原始Context
     * @return 应用了语言设置的新Context
     */
    public static Context applyLanguage(Context context) {
        String language = getLanguage(context);

        if (LANGUAGE_AUTO.equals(language)) {
            return context;
        }

        return updateContextLocale(context, language);
    }

    /**
     * 更新Context的Locale
     * @param context 原始Context
     * @param language 语言代码
     * @return 更新后的Context
     */
    private static Context updateContextLocale(Context context, String language) {
        Locale locale = getLocaleFromLanguage(language);

        Resources resources = context.getResources();
        Configuration config = new Configuration(resources.getConfiguration());

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            // Android 7.0+ 使用 LocaleList
            LocaleList localeList = new LocaleList(locale);
            LocaleList.setDefault(localeList);
            config.setLocales(localeList);
        } else {
            // Android 7.0 以下
            config.setLocale(locale);
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
            return context.createConfigurationContext(config);
        } else {
            resources.updateConfiguration(config, resources.getDisplayMetrics());
            return context;
        }
    }

    /**
     * 从语言代码获取Locale对象
     * @param language 语言代码
     * @return Locale对象
     */
    private static Locale getLocaleFromLanguage(String language) {
        switch (language) {
            case LANGUAGE_ZH:
                return Locale.SIMPLIFIED_CHINESE;
            case LANGUAGE_EN:
                return Locale.ENGLISH;
            case LANGUAGE_RU:
                return LOCALE_RUSSIAN;
            default:
                return Locale.getDefault();
        }
    }

    /**
     * 获取语言的显示名称
     * @param language 语言代码
     * @return 显示名称
     */
    public static String getLanguageDisplayName(String language) {
        switch (language) {
            case LANGUAGE_AUTO:
                return "Follow System";
            case LANGUAGE_ZH:
                return "简体中文";
            case LANGUAGE_EN:
                return "English";
            case LANGUAGE_RU:
                return "Русский";
            default:
                return language;
        }
    }

    /**
     * 获取所有支持的语言列表
     * @return 语言代码数组
     */
    public static String[] getSupportedLanguages() {
        return new String[]{
            LANGUAGE_AUTO,
            LANGUAGE_ZH,
            LANGUAGE_EN,
            LANGUAGE_RU
        };
    }
}

File: app/src/main/java/com/app/ralaunch/utils/LocaleManager.kt
package com.app.ralaunch.utils

import android.content.Context
import android.content.res.Configuration
import android.os.Build
import android.os.LocaleList
import com.app.ralaunch.shared.AppConstants
import com.app.ralaunch.shared.util.LocaleHelper
import com.app.ralaunch.shared.util.SupportedLanguage
import com.app.ralaunch.shared.util.LocaleManager as ILocaleManager
import java.util.Locale

/**
 * 多语言管理器 - Android 实现
 * 支持中文、英文等多种语言的动态切换
 * 
 * 实现 shared 模块的 LocaleManager 接口
 */
object LocaleManager : ILocaleManager {
    private const val PREFS_NAME = AppConstants.PREFS_NAME
    private const val KEY_LANGUAGE = "app_language"

    // 使用 shared 模块的常量
    const val LANGUAGE_AUTO = LocaleHelper.LANGUAGE_AUTO
    const val LANGUAGE_ZH = LocaleHelper.LANGUAGE_ZH
    const val LANGUAGE_EN = LocaleHelper.LANGUAGE_EN
    const val LANGUAGE_RU = LocaleHelper.LANGUAGE_RU
    const val LANGUAGE_ES = LocaleHelper.LANGUAGE_ES

    private val LOCALE_RUSSIAN = Locale("ru")
    private val LOCALE_SPANISH = Locale("es")
    
    // 用于存储当前 Context（弱引用场景下使用）
    private var currentLanguage: String = LANGUAGE_AUTO

    /**
     * 获取当前设置的语言
     */
    @JvmStatic
    fun getLanguage(context: Context): String {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_LANGUAGE, LANGUAGE_AUTO) ?: LANGUAGE_AUTO
    }

    /**
     * 设置语言
     */
    @JvmStatic
    fun setLanguage(context: Context, language: String) {
        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            .edit()
            .putString(KEY_LANGUAGE, language)
            .apply()
    }

    /**
     * 应用语言设置到Context
     */
    @JvmStatic
    fun applyLanguage(context: Context?): Context? {
        context ?: return null
        val language = getLanguage(context)
        return if (language == LANGUAGE_AUTO) context else updateContextLocale(context, language)
    }

    private fun updateContextLocale(context: Context, language: String): Context {
        val locale = getLocaleFromLanguage(language)
        val resources = context.resources
        val config = Configuration(resources.configuration)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            val localeList = LocaleList(locale)
            LocaleList.setDefault(localeList)
            config.setLocales(localeList)
        } else {
            config.setLocale(locale)
        }

        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
            context.createConfigurationContext(config)
        } else {
            @Suppress("DEPRECATION")
            resources.updateConfiguration(config, resources.displayMetrics)
            context
        }
    }

    private fun getLocaleFromLanguage(language: String): Locale = when (language) {
        LANGUAGE_ZH -> Locale.SIMPLIFIED_CHINESE
        LANGUAGE_EN -> Locale.ENGLISH
        LANGUAGE_RU -> LOCALE_RUSSIAN
        LANGUAGE_ES -> LOCALE_SPANISH
        else -> Locale.getDefault()
    }

    /**
     * 获取语言的显示名称
     */
    override fun getLanguageDisplayName(languageCode: String): String = when (languageCode) {
        LANGUAGE_AUTO -> "Follow System"
        LANGUAGE_ZH -> "简体中文"
        LANGUAGE_EN -> "English"
        LANGUAGE_RU -> "Русский"
        LANGUAGE_ES -> "Español"
        else -> languageCode
    }
    
    // JvmStatic 版本供 Java 调用
    @JvmStatic
    fun getDisplayName(language: String): String = getLanguageDisplayName(language)

    /**
     * 获取所有支持的语言列表（旧方法，兼容性保留）
     */
    @JvmStatic
    @Deprecated("使用 getSupportedLanguages(): List<SupportedLanguage>")
    fun getSupportedLanguagesArray(): Array<String> = arrayOf(
        LANGUAGE_AUTO,
        LANGUAGE_ZH,
        LANGUAGE_EN,
        LANGUAGE_RU,
        LANGUAGE_ES
    )

    // ==================== ILocaleManager 接口实现 ====================

    /**
     * 获取当前语言代码（接口实现）
     */
    override fun getCurrentLanguage(): String = currentLanguage

    /**
     * 设置语言（接口实现）
     */
    override fun setLanguage(languageCode: String) {
        currentLanguage = languageCode
    }

    /**
     * 获取支持的语言列表（接口实现）
     */
    override fun getSupportedLanguages(): List<SupportedLanguage> {
        return SupportedLanguage.primaryLanguages()
    }
}
